collector/slave_status.go

```diff
@@ -77,16 +79,16 @@ func ScrapeSlaveStatus(db *sql.DB, ch chan<- prometheus.Metric) error {
 			if value, ok := parseStatus(*scanArgs[i].(*sql.RawBytes)); ok { // Silently skip unparsable values.
 				ch <- prometheus.MustNewConstMetric(
 					prometheus.NewDesc(
-						prometheus.BuildFQName("mysql", slaveStatus, strings.ToLower(col)),
+						prometheus.BuildFQName(namespace, slaveStatus, strings.ToLower(col)),
 						"Generic metric from SHOW SLAVE STATUS.",
 						[]string{"master_uuid", "master_host", "channel_name"},
 						nil,
 					),
 					prometheus.UntypedValue,
 					value,
-					columnValue(scanArgs, slaveCols, "Master_UUID"),
-					columnValue(scanArgs, slaveCols, "Master_Host"),
-					columnValue(scanArgs, slaveCols, "Channel_Name"),
+					columnValue(scanArgs, masterUUIDIndex),
```

Comment:
I was more thinking why not save the output of `columnValue()` after you assign `scanArgs`.  This way you don't need to execute the function for every loop over `slaveCols`